src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HashProviderDispenser.Browser.cs

```diff
@@ -32,9 +30,16 @@ public static class OneShotHashProvider
                 ReadOnlySpan<byte> source,
                 Span<byte> destination)
             {
-                HashProvider provider = CreateMacProvider(hashAlgorithmId, key);
-                provider.AppendHashData(source);
-                return provider.FinalizeHashAndReset(destination);
+                if (CanUseSubtleCryptoImpl)
+                {
+                    return HMACNativeHashProvider.MacDataOneShot(hashAlgorithmId, key, source, destination);
+                }
+                else
+                {
+                    HashProvider provider = CreateMacProvider(hashAlgorithmId, key);
```

Comment:
Since we are clearing the key on `Dispose` (and only if dispose, not finalizer) I think the `HashProvider` should be in a `using`.